public/js/Game.js

```diff
@@ -0,0 +1,107 @@
+ function Game(){
+   this.scoreTotal = 0;
+   this.currentFrame = 1;
+   this.currentRoll = 1;
+   this.lastRollScore = 0;
+   this.pins = 10;
+   this.scoreMode = "normal";
+   this.gameOver = false;
+ }
+
+Game.prototype.hits = function() {
+  return Math.floor(Math.random() * (this.pins + 1));
+};
+
+
+Game.prototype.roll = function() {
+  var hits = this.hits();
+  this.score(hits);
+  this.resetStatus(hits);
+  if (!this.gameOver) {
+    this.resetPins(hits);
+    this.resetMode(hits, this.scoreMode);
+    this.resetRolls(hits);
+    this.lastRollScore = hits;
+  }
+};
+
+Game.prototype.score = function(hits) {
+  if (this.scoreMode === "normal") {
```

Comment:
Game feels like a large unit.  Could some of its responsibilities be moved to another unit? (For example, by refactoring to keep every roll that has happened and then refactoring some of the logic for managing the scoring of frames to a new unit?)